reV/supply_curve/aggregation.py

```diff
@@ -354,228 +194,58 @@ def _check_files(self):
-        gen_fpath : str
-            Filepath to .h5 reV generation output results.
-        res_class_dset : str | None
-            Dataset in the generation file dictating resource classes.
-            None if no resource classes.
-        res_class_bins : list | None
-            List of two-entry lists dictating the resource class bins.
-            None if no resource classes.
-        cf_dset : str
-            Dataset name from f_gen containing capacity factor mean values.
-        lcoe_dset : str
-            Dataset name from f_gen containing LCOE mean values.
-
-        Returns
-        -------
-        res_data : np.ndarray | None
-            Extracted resource data from res_class_dset
-        res_class_bins : list
-            List of resouce class bin ranges.
-        cf_data : np.ndarray | None
-            Capacity factor data extracted from cf_dset in gen
-        lcoe_data : np.ndarray | None
-            LCOE data extracted from lcoe_dset in gen
-        offshore_flags : np.ndarray | None
-            Array of offshore boolean flags if available from wind generation
-            data. None if offshore flag is not available.
-        """
-
-        if res_class_dset is None:
-            res_data = None
-            res_class_bins = [None]
-        else:
-            res_data = gen[res_class_dset]
-
-        if res_class_bins is None:
-            res_class_bins = [None]
-
-        if cf_dset in gen.datasets:
-            cf_data = gen[cf_dset]
-        else:
-            cf_data = None
-            w = ('Could not find cf dataset "{}" in '
-                 'generation file: {}'
-                 .format(cf_dset, gen_fpath))
-            logger.warning(w)
-            warn(w, OutputWarning)
-
-        if lcoe_dset in gen.datasets:
-            lcoe_data = gen[lcoe_dset]
-        else:
-            lcoe_data = None
-            w = ('Could not find lcoe dataset "{}" in '
-                 'generation file: {}'
-                 .format(lcoe_dset, gen_fpath))
-            logger.warning(w)
-            warn(w, OutputWarning)
-
-        if 'offshore' in gen.meta:
-            offshore_flag = gen.meta['offshore'].values
-        else:
-            offshore_flag = None
-
-        return res_data, res_class_bins, cf_data, lcoe_data, offshore_flag
-
-    @staticmethod
-    def _serial_summary(excl_fpath, gen_fpath, tm_dset, gen_index,
-                        excl_dict=None, res_class_dset=None,
-                        res_class_bins=None, cf_dset='cf_mean-means',
-                        lcoe_dset='lcoe_fcr-means', data_layers=None,
-                        resolution=64, power_density=None, friction_fpath=None,
-                        friction_dset=None, gids=None,
-                        area_filter_kernel='queen', min_area=None, args=None,
-                        ex_area=0.0081, close=False, check_excl_layers=False):
+    @abstractmethod
+    def run_serial(sc_point_method, excl_fpath, tm_dset,
```

Comment:
I learned there's actually a `abstractstaticmethod` so you don't have to stack abtractmethod and staticmethod

Reply:
cool!